lib/eval.ml

```diff
@@ -1241,19 +1399,23 @@ module MakeInterpreter (T : Target) = struct
   (*----------------------------------------------------------------------------*)
 
   and eval_parser (ctrl : ctrl) (env : env) (st : state) (params : TypeParameter.t list)
-      (args : Expression.t option list) (vs : (string * value) list)
+      (args : Expression.t option list) (pscope : env) (ls : (string * loc) list)
       (locals : Declaration.t list) (states : Parser.state list) : env * state * signal =
-    let (penv, st', s) = copyin ctrl env st params args in
+    (* TODO: incorporate closure environment *)
```

Comment:
The closure environment is `pscope`, right? I don't see it being used

Reply:
there was a point of semantic confusion about this that we wanted to discuss either with you or nate and we didnt want to block this PR on it

Reply:
There was a point of confusion with semantics here that we wanted to discuss before inserting the use of pscope. We also didn't want to block this PR on that change since there aren't any cases failing because of it.